packages/mui-material/src/Fab/Fab.js

```diff
@@ -12,14 +12,7 @@ const useUtilityClasses = (ownerState) => {
   const { color, variant, classes, size } = ownerState;
 
   const slots = {
-    root: [
-      'root',
-      variant,
-      `size${capitalize(size)}`,
-      color === 'inherit' && 'colorInherit',
-      color === 'primary' && 'primary',
-      color === 'secondary' && 'secondary',
-    ],
+    root: ['root', variant, `size${capitalize(size)}`, color],
```

Comment:
We still need the condition for `colorInherit` otherwise this would be a breaking change.
```suggestion
    root: ['root', variant, `size${capitalize(size)}`, ownerState.color === 'inherit' ? 'colorInherit': color],
```

Maybe something like this?